### PR TITLE
Remove django 4.0 deprecations

### PR DESCRIPTION
**Pull request by ckleemann**
With this pr the deprecation warnings introduced in Django 3.1 are removed. 
Also the travis ci build matrix was updated to current supported Django version. In addition deprecated keys in the travis-ci.yml where removed or replaced by there successors.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,6 +42,7 @@ before_script:
   - ${GNUPG_BINARY:-gpg} --version
   - mkdir gpg_keyring
 script:
+  - export PYTHONWARNINGS=all
```
Comment (blag) [inline .travis.yml]: I could be wrong, but I don't think `all` is a valid value for this environment variable.
See here: https://docs.python.org/3/using/cmdline.html#envvar-PYTHONWARNINGS
Can you point me to where `all` is supported?

**Review by blag — CHANGES_REQUESTED** [review]
Please check valid values as noted.

Otherwise this is looking good. Thanks for the contribution!

File: .travis.yml (hunk 1 above)
Comment (ckleemann) [reply .travis.yml]: It seams as this has changed in Python 3.7, which I was not aware of until now. In Python 3.6 there [is still the value ```all```.](https://docs.python.org/3.6/using/cmdline.html#cmdoption-w) There is a mention about it in the [Python 3.7 release notes.](https://docs.python.org/3/whatsnew/3.7.html#warnings) As ```always``` is also supported, but not documented, in Python 3.6 I will change it to ```always```.

Comment (blag) [reply .travis.yml]: That situation is...not great. Using `always` sounds like the way to go. Thanks for changing it!

**Comment by blag** [comment]
This is now available in [version `1.3.0` on PyPI](https://pypi.org/project/django-secure-mail/1.3.0/). Thank you!

**Pull request merged**